addons/project/models/res_partner.py

```diff
@@ -80,3 +80,30 @@ def action_view_tasks(self):
         else:
             action['domain'] = search_domain
         return action
+
+    @api.model
+    def get_mention_suggestions_from_task(self, task_id, search, limit=8):
+        """Return 'limit'-first partners' such that the name or email matches a 'search' string.
+        Prioritize partners that are also (internal) users, and then extend the research to all partners.
+        Only followers of the given task or followers of its project are returned.
+        The return format is a list of partner data (as per returned by `mail_partner_format()`).
+        """
+        task_sudo = self.env['project.task'].sudo().search([('id', '=', task_id)])
+        shared_project_sudo = task_sudo.project_id
+        shared_project = shared_project_sudo.with_user(self.env.user)
+        if not (shared_project and shared_project._check_project_sharing_access()):
+            return []
+        try:
+            shared_project.check_access_rights('read')
+            shared_project.check_access_rule('read')
+        except AccessError:
+            return []
+        followers = shared_project_sudo.message_follower_ids + task_sudo.message_follower_ids
```

Comment:
If we apply all the other comments, I made then indeed here we need to fetch `follower_ids` in sudo to make sure we get all followers. 🙂
Moreover, we could do the union (`|` operator) to avoid having duplicating records inside the recordset since the followers inside the task could also be followers of the project shared. 🙂 

```suggestion
        followers = project.sudo().message_follower_ids | task.sudo().message_follower_ids
```